standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -7046,6 +7047,52 @@ public void abort_txns(AbortTxnsRequest rqst) throws TException {
 
     @Override
     public void commit_txn(CommitTxnRequest rqst) throws TException {
+      // in replication flow, the write notification log table will be updated here.
+      if (rqst.isSetWriteEventInfos()) {
+        long targetTxnId = getTxnHandler().getTargetTxnId(rqst.getReplPolicy(), rqst.getTxnid());
+        if (targetTxnId < 0) {
+          //looks like a retry
+          return;
+        }
+        for (WriteEventInfo writeEventInfo : rqst.getWriteEventInfos()) {
```

Comment:
Can we have some comment here as below logic is not clear. Also, can we have this logic in CommitTxnHandler where we populate the WriteEventsInfo list as it looks clean there?

Reply:
add a comment explaining the logic ..no need to change code for now

Reply:
done